issues with Findlibxmp-lite.cmake

Tested this with SDL2 branch, but they are the same in SDL3 too.

With current source tree, running cmake with `-DSDL2MIXER_MOD_XMP_LITE=ON`
I get:
```
-- Using system libxmp-lite
CMake Warning (dev) at /opt/cmake3.18/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args`
  (libxmp_lite) does not match the name of the calling package (libxmp-lite).
  This can lead to problems in calling code that expects `find_package`
  result variables (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Findlibxmp-lite.cmake:17 (find_package_handle_standard_args)
  CMakeLists.txt:664 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find libxmp_lite (missing: libxmp_lite_LIBRARY libxmp_lite_INCLUDE_PATH) 
CMake Error at cmake/PrivateSdlFunctions.cmake:141 (get_target_property):
  get_target_property() called with non-existent target
  "libxmp-lite::libxmp-lite".
Call Stack (most recent call first):
  cmake/PrivateSdlFunctions.cmake:151 (get_actual_target)
  CMakeLists.txt:686 (target_get_dynamic_library)


-- Dynamic libxmp-lite: 
[...]
-- Configuring incomplete, errors occurred!
```

If I apply the following change, it succeeds (note that `-shared` is
NOT present in Findlibxmp.cmake, only in the lite version..)
```diff
diff --git a/cmake/Findlibxmp-lite.cmake b/cmake/Findlibxmp-lite.cmake
index 573409f..a7d97f2 100644
--- a/cmake/Findlibxmp-lite.cmake
+++ b/cmake/Findlibxmp-lite.cmake
@@ -1,11 +1,11 @@
 include(FindPackageHandleStandardArgs)
 
 find_library(libxmp_lite_LIBRARY
-    NAMES xmp
+    NAMES xmp-lite
 )
 
 find_path(libxmp_lite_INCLUDE_PATH
-    NAMES xmp.h
+    NAMES libxmp-lite/xmp.h xmp.h
 )
 
 set(libxmp_lite_COMPILE_OPTIONS "" CACHE STRING "Extra compile options of libxmp_lite")
@@ -21,7 +21,7 @@ find_package_handle_standard_args(libxmp_lite
 if(libxmp_lite_FOUND)
     if(NOT TARGET libxmp-lite::libxmp-lite)
         add_library(libxmp-lite::libxmp-lite UNKNOWN IMPORTED)
-        set_target_properties(libxmp_lite::libxmp_lite-shared PROPERTIES
+        set_target_properties(libxmp-lite::libxmp-lite PROPERTIES
             IMPORTED_LOCATION "${libxmp_lite_LIBRARY}"
             INTERFACE_INCLUDE_DIRECTORIES "${libxmp_lite_INCLUDE_PATH}"
             INTERFACE_COMPILE_OPTIONS "${libxmp_lite_COMPILE_OPTIONS}"
```

... BUT, still with a developer warning:
```
- Using system libxmp-lite
CMake Warning (dev) at /opt/cmake3.18/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args`
  (libxmp_lite) does not match the name of the calling package (libxmp-lite).
  This can lead to problems in calling code that expects `find_package`
  result variables (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Findlibxmp-lite.cmake:17 (find_package_handle_standard_args)
  CMakeLists.txt:664 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found libxmp_lite: /home/sezero/z2/lib/libxmp-lite.so  
```

I did not apply my patch above: Leaving things to @madebr.
P.S.: With the patch above, cmake itself finds the xmp.h header but the
compiler obviously does not.

--- a/cmake/Findlibxmp-lite.cmake
+++ b/cmake/Findlibxmp-lite.cmake
@@ -1,11 +1,12 @@
 include(FindPackageHandleStandardArgs)
 
 find_library(libxmp_lite_LIBRARY
-    NAMES xmp
+    NAMES xmp-lite
 )
 
 find_path(libxmp_lite_INCLUDE_PATH
     NAMES xmp.h
+    PATH_SUFFIXES libxmp-lite
 )
 
 set(libxmp_lite_COMPILE_OPTIONS "" CACHE STRING "Extra compile options of libxmp_lite")
@@ -14,14 +15,14 @@ set(libxmp_lite_LINK_LIBRARIES "" CACHE STRING "Extra link libraries of libxmp_l
 
 set(libxmp_lite_LINK_FLAGS "" CACHE STRING "Extra link flags of libxmp_lite")
 
-find_package_handle_standard_args(libxmp_lite
+find_package_handle_standard_args(libxmp-lite
     REQUIRED_VARS libxmp_lite_LIBRARY libxmp_lite_INCLUDE_PATH
 )
 
 if(libxmp_lite_FOUND)
     if(NOT TARGET libxmp-lite::libxmp-lite)
         add_library(libxmp-lite::libxmp-lite UNKNOWN IMPORTED)
-        set_target_properties(libxmp_lite::libxmp_lite-shared PROPERTIES
+        set_target_properties(libxmp-lite::libxmp-lite PROPERTIES
             IMPORTED_LOCATION "${libxmp_lite_LIBRARY}"
             INTERFACE_INCLUDE_DIRECTORIES "${libxmp_lite_INCLUDE_PATH}"
             INTERFACE_COMPILE_OPTIONS "${libxmp_lite_COMPILE_OPTIONS}"

